test/innerJoin.js

```diff
@@ -0,0 +1,25 @@
+var R = require('..');
+var eq = require('./shared/eq');
+
+
+var a = {id: 1, name: 'a'};
+var b = {id: 2, name: 'b'};
+var c = {id: 3, name: 'c'};
+var f = R.innerJoin(function(r, id) { return r.id === id; });
+
+
+describe('innerJoin', function() {
+
+  it('only returns elements from the first list', function() {
+    eq(f([a, b, c], []), []);
+    eq(f([a, b, c], [1]), [a]);
+    eq(f([a, b, c], [1, 2]), [a, b]);
+    eq(f([a, b, c], [1, 2, 3]), [a, b, c]);
+    eq(f([a, b, c], [1, 2, 3, 4]), [a, b, c]);
+  });
+
+  it('preserves the ordering of the first list', function() {
+    eq(f([c, a, b], [1, 2, 3]), [c, a, b]);
```

Comment:
> We're using an ordered collection. 

but *why* is that? is that fact *intrinsic* to this function? or just we have at hand? if the *intent* of this function is *only ordered lists* then this test makes sense. If the intent is relations generally, but for now all we have to implement it is Arrays, then this test may be unnecessarily prescriptive. That's all. Let's move on.